agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -2367,6 +2368,33 @@ bool backhaul_manager::handle_1905_topology_query(ieee1905_1::CmduMessageRx &cmd
         }
     }
 
+    for (const auto &neighbor_device : m_1905_neighbor_devices) {
```

Comment:
This is not entirely correct: there should be one tlv1905NeighborDevice per local interface, not per neighbour. And for each of those, *all* the neighbours visible from that interface should be added.

So in practice, this means inverting the m_1905_neighbor_devices map into a map-of-vector-of-neighbor-device.

Reply:
Ok, let's make this clear. Because clearly it is not clear to me.
I'm filling the container from Discovery message which has:
1905.1 AL MAC address type TLV and MAC address type TLV.
1905.1 AL MAC address type is the neighbor (peer?) mac?
and MAC address type TLV is the local iface_mac (or not?)?

Anyway, you say that I should store the information like this?
```cpp
    /*
     * key:     Local interface mac
     * value:   s1905NeighborDevice
     * Devices are being added to the list when receiving a 1905.1 Topology Discovery message from
     * an unknown 1905.1 device. Every 1905.1 device shall send this message every 60 seconds, and
     * we update the time stamp in which the message is received.
     */
    struct s1905NeighborDevice {
        std::chrono::steady_clock::time_point last_topology_discovery;
        std::list<sMacAddr> al_mac;
        bool bridge_exist; // Not sure from where to take this information
    };
    std::unordered_map<sMacAddr, s1905NeighborDevice> m_1905_neighbor_devices;

```